components/sup/src/manager/mod.rs

```diff
@@ -303,11 +313,7 @@ impl Manager {
         loop {
             let next_check = SteadyTime::now() + TimeDuration::milliseconds(1000);
             if self.check_for_incoming_signals() {
-                let mut services = self.services.write().expect("Services lock is poisend!");
-                for service in services.drain(..) {
-                    self.remove_service(&service)?;
-                }
-                outputln!("Habitat thanks you - shutting down!");
+                self.shutdown();
```

Comment:
Would it be worth considering call to this on `Drop` of `Manager`? It would have to be idempotent--and maybe that's overkill